### PR TITLE
interfaces: Improve error message when targeting LF 1.14

### PR DESCRIPTION
**Pull request by sofiafaro-da**
This PR improves the error message you get when targeting
an LF version that doesn't support interfaces.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs [hunk 1]
```diff
@@ -315,17 +315,17 @@ convertRationalNumericMono :: Env -> Integer -> Integer -> Integer -> ConvertM L
 convertRationalNumericMono env scale num denom
     | scale < 0 || scale > fromIntegral numericMaxScale =
         unsupported
-            ("Tried to construct value of type Numeric " ++ show scale ++ ", but scale is out of bounds. Scale must be between 0 through 37, not " ++ show scale)
+            ("Tried to construct value of type Numeric " ++ show scale ++ ", but scale is out of bounds. Scale must be between 0 through 37, not " ++ show scale ++ ".")
```
Comment (cocreature) [inline compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs:318]: Can you write pointfree Haskell if you end your error messages with a dot?

**Review by akrmn — APPROVED** [review]
Thanks!

**Review by cocreature — APPROVED** [review]
Thank you!

File: compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs (hunk 1 above)
Comment (sofiafaro-da) [reply compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs:318]: 🤔

**Pull request merged**